### ISSUE TITLE
Allowed memory size of 1610612736 bytes exhausted DependencyResolver/Solver.php on line 223

### ISSUE THREAD

**Issue by williamdes** [composer/composer#8953]
### Story

On a project using GitHub actions, circle CI, different dependencies we started to see php errors on installs or updates.

Dont get me wrong, on 2 of June 2020 the CI did pass and on the 3 of June 2020 the same CI did not pass.

Sure, I updated to the latest composer version and nothing did solve the memory problem.

What did go wrong ?

We had to add `COMPOSER_MEMORY_LIMIT=-1`

I get the following output:

```
Fatal error: Allowed memory size of 1610612736 bytes exhausted (tried to allocate 67108864 bytes) in phar:///usr/bin/composer/src/Composer/DependencyResolver/Solver.php on line 223
```

And I expected this to happen:

No errors

**Comment by williamdes** [comment]
```
bash-5.0# composer update -vvvv
Reading ./composer.json
Loading config file ./composer.json
Checked CA file /etc/ssl/certs/ca-certificates.crt: valid
Executing command (/var/www): git branch --no-color --no-abbrev -v
Failed to initialize global composer: Composer could not find the config file: /root/.composer/composer.json
To initialize a project, please create a composer.json file as described in the https://getcomposer.org/ "Getting Started" section
Reading /var/www/vendor/composer/installed.json
Loading plugin PackageVersions\Installer
Running 1.10.6 (2020-05-06 10:28:10) with PHP 7.2.31 on Linux / 5.3.0-55-generic
Loading composer repositories with package information
Downloading https://repo.packagist.org/packages.json
Writing /root/.composer/cache/repo/https---repo.packagist.org/packages.json into cache
Updating dependencies (including require-dev)
Reading /root/.composer/cache/repo/https---repo.packagist.org/p-provider-2013.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/p-provider-2014.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/p-provider-2015.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/p-provider-2016.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/p-provider-2017.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/p-provider-2018.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/p-provider-2019.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/p-provider-2019-07.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/p-provider-2019-10.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/p-provider-2020-01.json from cache
Downloading http://repo.packagist.org/p/provider-2020-04%240e3b18405bc291f1e0fd8b6e08bbf48fc962a6f812d1ed00235f6d6cbecd5742.json
Writing /root/.composer/cache/repo/https---repo.packagist.org/p-provider-2020-04.json into cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/p-provider-archived.json from cache
Downloading http://repo.packagist.org/p/provider-latest%2493d9bfe9109289622549f017764f939e2db1fa5dede3db27a4c1358ad8a6674f.json
Writing /root/.composer/cache/repo/https---repo.packagist.org/p-provider-latest.json into cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-illuminate$contracts.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-illuminate$database.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-illuminate$support.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-astrotomic$laravel-translatable.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-psr$container.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-psr$simple-cache.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-illuminate$container.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$console.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-doctrine$inflector.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-nesbot$carbon.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-voku$portable-ascii.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$polyfill-mbstring.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$polyfill-php73.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$service-contracts.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$polyfill-php80.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$string.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$translation.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-kylekatarnls$update-helper.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$polyfill-ctype.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$polyfill-intl-grapheme.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$polyfill-intl-normalizer.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$contracts.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$translation-contracts.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-psr$cache.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-guzzlehttp$guzzle.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-guzzlehttp$promises.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-guzzlehttp$psr7.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-mtdowling$jmespath.php.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-aws$aws-sdk-php.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$polyfill-intl-idn.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-psr$http-message.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-ralouphie$getallheaders.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$polyfill-php72.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-illuminate$routing.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-illuminate$session.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-maximebf$debugbar.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$debug.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$finder.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-barryvdh$laravel-debugbar.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-illuminate$http.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-illuminate$pipeline.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$http-foundation.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$http-kernel.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$routing.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-illuminate$filesystem.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-psr$log.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$var-dumper.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$mime.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$event-dispatcher.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$polyfill-php56.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$error-handler.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$polyfill-php70.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$deprecation-contracts.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$event-dispatcher-contracts.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$polyfill-util.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-paragonie$random-compat.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-psr$event-dispatcher.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-barryvdh$reflection-docblock.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-composer$composer.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-doctrine$dbal.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-illuminate$console.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-barryvdh$laravel-ide-helper.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-composer$ca-bundle.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-composer$semver.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-composer$spdx-licenses.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-composer$xdebug-handler.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-justinrainbow$json-schema.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-seld$jsonlint.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-seld$phar-utils.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$filesystem.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$process.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-seld$cli-prompt.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-doctrine$cache.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-doctrine$event-manager.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-knplabs$knp-snappy.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-barryvdh$laravel-snappy.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-monolog$monolog.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-brozot$laravel-fcm.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-danielstjules$stringy.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-clue$stream-filter.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-composer$package-versions-deprecated.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-daltcore$lara-pdf-merger.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-dnoegel$php-xdg-base-dir.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-doctrine$lexer.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-doctrine$annotations.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-doctrine$instantiator.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-dragonmantank$cron-expression.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-editorconfig-checker$editorconfig-checker.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-egulias$email-validator.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-eluceo$ical.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-erusev$parsedown.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-ezyang$htmlpurifier.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-facebook$graph-sdk.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-filp$whoops.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-fzaninotto$faker.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-giggsey$locale.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-giggsey$libphonenumber-for-php.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-greggilbert$recaptcha.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-hamcrest$hamcrest-php.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-psr$http-factory.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-http-interop$http-factory-guzzle.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-intervention$image.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-illuminate$validation.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-intervention$validation.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-illuminate$translation.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$polyfill-php54.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$polyfill-php55.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-ircmaxell$password-compat.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-jakub-onderka$php-console-color.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-jakub-onderka$php-console-highlighter.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-jaybizzle$crawler-detect.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-jean85$pretty-package-versions.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-mobiledetect$mobiledetectlib.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-jenssegers$agent.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-jenssegers$date.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-phpunit$phpunit.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-johnkary$phpunit-speedtrap.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-myclabs$deep-copy.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-phar-io$manifest.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-phar-io$version.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-phpspec$prophecy.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-phpunit$php-code-coverage.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-phpunit$php-file-iterator.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-phpunit$php-text-template.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-phpunit$php-timer.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-sebastian$comparator.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-sebastian$diff.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-sebastian$environment.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-sebastian$exporter.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-sebastian$global-state.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-sebastian$object-enumerator.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-sebastian$resource-operations.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-sebastian$version.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-sebastian$type.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-phpunit$php-invoker.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-sebastian$code-unit.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-phpdocumentor$reflection-docblock.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-sebastian$recursion-context.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-phpunit$php-token-stream.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-sebastian$code-unit-reverse-lookup.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-theseer$tokenizer.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-sebastian$object-reflector.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-dflydev$markdown.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-phpdocumentor$reflection-common.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-phpdocumentor$type-resolver.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-webmozart$assert.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-league$flysystem.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-opis$closure.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-ramsey$uuid.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-swiftmailer$swiftmailer.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-tijsverkoyen$css-to-inline-styles.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-vlucas$phpdotenv.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-laravel$framework.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$polyfill-iconv.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$css-selector.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-phpoption$phpoption.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-psy$psysh.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-laravel$tinker.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-nikic$php-parser.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-illuminate$view.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-laravelcollective$html.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-illuminate$events.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-lcobucci$jwt.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-league$flysystem-aws-s3-v3.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-league$fractal.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-league$iso3166.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-phpoffice$phpspreadsheet.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-maatwebsite$excel.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-markbaker$complex.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-markbaker$matrix.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-maennchen$zipstream-php.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-myclabs$php-enum.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-vimeo$psalm.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-amphp$amp.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-amphp$byte-stream.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-felixfbecker$advanced-json-rpc.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-felixfbecker$language-server-protocol.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-netresearch$jsonmapper.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-ocramius$package-versions.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-openlss$lib-array2xml.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-webmozart$glob.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-webmozart$path-util.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-mattketmo$email-checker.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-mockery$mockery.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-namshi$jose.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-nette$di.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-nette$utils.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-nette$bootstrap.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-nette$neon.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-nette$php-generator.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-nette$robot-loader.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-nette$schema.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-nette$finder.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-nunomaduro$collision.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-orchestra$testbench.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-phpstan$phpstan.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-nunomaduro$larastan.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-orchestra$testbench-core.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-phpstan$phpdoc-parser.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-illuminate$cache.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-illuminate$config.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-omniphx$forrest.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-php-http$httplug.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-php-http$message.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-php-http$message-factory.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$options-resolver.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-php-http$client-common.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-psr$http-client.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-php-http$promise.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-php-http$discovery.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-php-http$guzzle6-adapter.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-phplang$scope-exit.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-ronanguilloux$isocodes.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-pixelpeter$laravel5-isocodes-validation.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-predis$predis.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-propaganistas$laravel-phone.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-sentry$sentry.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-sentry$sdk.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-php-http$async-client-implementation.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-psr$http-message-implementation.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$polyfill-uuid.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-sentry$sentry-laravel.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-squizlabs$php-codesniffer.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-slevomat$coding-standard.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-swaggest$json-diff.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-swaggest$json-schema.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$polyfill-intl-icu.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$intl.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-symfony$yaml.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-therezor$laravel-transactional-jobs.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-illuminate$auth.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-tymon$jwt-auth.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-illuminate$queue.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-zircote$swagger-php.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-julien-c$iso3166.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-php-http$curl-client.json from cache
Reading /root/.composer/cache/repo/https---repo.packagist.org/provider-zendframework$zend-diactoros.json from cache

Fatal error: Allowed memory size of 1610612736 bytes exhausted (tried to allocate 67108864 bytes) in phar:///usr/bin/composer/src/Composer/DependencyResolver/Solver.php on line 223

Check https://getcomposer.org/doc/articles/troubleshooting.md#memory-limit-errors for more info on how to handle out of memory errors.bash-5.0# 
```

**Comment by williamdes** [comment]
@Seldaek `provider-zendframework$zend-diactoros.json` seems to be an issue
![image](https://user-images.githubusercontent.com/7784660/83891216-1c756f00-a74d-11ea-8884-6d9358505fd2.png)

You can clearly see the RAM usage 

## With no limit

![image](https://user-images.githubusercontent.com/7784660/83891390-5e061a00-a74d-11ea-93c1-3d4e74de68d9.png)

**Comment by Seldaek** [comment]
Please try switching to composer 2.0 by adding `composer self-update --preview` to your build. It should help a lot, unless you have issues with plugins. We are not going to fix any of this in 1.x..

**Comment by williamdes** [comment]
No issues after the upgrade, I understand the policy and think that we are avoiding the root cause that seems to be caused by some recent change somewhere.

Thank you for the tips :)

**Comment by jrfnl** [comment]
I just ran into the same thing and upgrading to Composer 2.0 isn't an option as the project uses a Composer plugin which isn't compatible (yet) with Composer 2.0.

After some trial and error, largely based on info from when the CI build was still passing, I've found that doing a `composer self-update 1.9.3` fixes it for us.

I haven't analyzed which version of Composer introduced the error between 1.9.3 and the last 1.x release, but hopefully this info may help someone else running into this.
